draft-normington-jsonpath-latest.xml

```diff
@@ -384,50 +384,115 @@ array-slice = [ integer ] ws %x3A ws [ integer ]
-                        in which case it selects the same element as the non-negative index <tt>j + len</tt>.
-                     </t>
-                     <t>An array slice selects any elements of an array which are indexed by the members of the slice,
-                        as defined below, in the same order as the slice.
-                        It selects no nodes from a node which is not an array.
-                     </t>
-                     <t>The members of a slice depend on whether the step value is positive, negative, or zero as follows. If the step value is
-                        omitted, it defaults to <tt>1</tt>.
-                     </t>
-                     <t>For positive integer <tt>step</tt> and integers <tt>start</tt> and <tt>end</tt>, the array slice
-                        <tt>start:end:step</tt> consists of the indices starting at <tt>start</tt> and increasing repeatedly
-                        by <tt>step</tt> while the resultant index is less than <tt>end</tt>.
-                        The indices of the slice are the same as the values of i iterated over by this <tt>for</tt> loop in C:
-                     </t>
-                     <sourcecode type="c">
-for (i = start; i &lt; end; i = i + step) {
-  ...
-}
+                     <t>Array slicing is inspired by the behaviour of the <tt>Array.prototype.slice</tt> method
+                        of the JavaScript language as defined by ECMA-262, with the addition of the <tt>step</tt>
+                        parameter, which is inspired by the Python slice expression. This section will fully
+                        specify its behaviour, without dependencies on externally defined behaviour.
+                        Array indexing is defined in terms of the slicing operation.</t>
+
+                     <t>A slice expression selects a subset of the elements of the input array, in the same order
+                        as the array or the reverse order, depending on the sign of the <tt>step</tt> parameter.</t>
+
+                     <t>A slice is defined by the two slice parameters, <tt>start</tt> and <tt>end</tt>,
+                        and an iteration delta, <tt>step</tt>. Each of these parameters is optional.
+                        <tt>len</tt> is the length of the input array.</t>
+
+                     <t>The default value for <tt>step</tt> is <tt>1</tt>.
+                        The default values for <tt>start</tt> and <tt>end</tt> depend on the sign of <tt>step</tt>,
+                        as defined in the following pseudocode:</t>
+
+                     <sourcecode type="pseudocode">
+IF step &gt;= 0 THEN
+
+  IF start is omitted THEN
+    start = 0
+  END IF
+  IF end is omitted THEN
+    end = len
+  END IF
+
+ELSE
+
+  IF start is omitted THEN
+    start = len - 1
+  END IF
+  IF end is omitted THEN
+    end = -len - 1
+  END IF
+
+END IF
                      </sourcecode>
```

Comment:
This is fine, but a table would have been equally easy to follow.

Reply:
thanks for the suggestion, rewrote as a table